### ISSUE TITLE
Fail fast when view isn't found

### ISSUE THREAD

**Issue by lyahdav** [indiegogo/KIF-Kiwi#6]
It's a bit annoying when a view with a certain accessibility label couldn't be found so I know that specific test will fail, but it keeps running the rest of the test. It makes things very slow on CI especially when there is no one controlling the build. Is there a way to cause tests to fail fast and have them stop if any failure occurs?

**Comment by rciobanu-omnifone** [comment]
Also interested in this problem. Dear maintainers, any thoughts?

**Comment by lyahdav** [comment]
Until there's an official fix I hacked this in by adding a call to exit(1) at the end of the failWithException method in KWExample (KIFTester) category.

**Comment by rciobanu-omnifone** [comment]
@lyahdav, thanks for sending me the right way. 
I think it would be better to use ``[exception raise];`` instead of ``exit(1)`` in the ``if (stop && [[exception reason] rangeOfString...`` block in order to fail only one test.
Assuming the tests are properly separated, then the others should continue as planned and shouldn't be influenced by the failing one.

**Comment by glentregoning** [comment]
Hi @lyahdav thanks for raising this. I'll have a look over your PR shortly.

**Comment by rciobanu-omnifone** [comment]
@glentregoning any news on this?

**Comment by glentregoning** [comment]
@rciobanu-omnifone sorry for the big delay looking at this. 

I definitely share the cascading test case pain that you and @lyahdav's have.

My only concern with this proposal is that I'm not sure in every case the best action is to fail fast. 

In my suites I observe that often there are two kinds of failures when a view is not found:
 1.  the failure which makes the rest of the assertions in the test invalid and likely to cascade through the rest of the assertions in the test case  [Your case]
    - Typically will happen on for a `[tester tapViewWithAccessibilityLabel:"Some label"]`
 2.  the failure in which an element is missing, but its absence doesn't affect following assertions. 
    - For example a `[tester waitForViewWithAccessibilityLabel:@"some view"]` which verifies state of a certain view, but don't impact following assertions. 

As a result I was thinking this should perhaps be an opt in capability. 

For reference I had a look over a how XCTest approaches this scenario (I know not always the best reference), but they provide this capability thorough the `-[XCTestCase continueAfterFailure:]` method.

As it turns out this opt in capability is also available in KIF & KIF-Kiwi. 

You can enable this capability within a given Kiwi spec by setting `self.continueAfterFailure = NO;`. The advantage of this approach is that it appears that you can enable / disable at whatever scope you desire within a single spec. There doesn't appear to be any global way to do this at the moment. 

Do you mind giving this a try and seeing it this works for your cases?

**Comment by rciobanu-omnifone** [comment]
@glentregoning 
Hi, your big delay is child's play compared to my big delay! :)
I'm so sorry (wow, 5 months!). Dunno how I missed the notification from Github.

I've tried to repro the issue with the latest KIF-Kiwi version and wasn't able to. Something must have changed in the mean time.
I thank you for the detailed response. Happy to start using the main branch again. I'll cancel the pull request.
